azerothcore/azerothcore-wotlk
# fix(DB): Add some scourge invasion gobjects/creatures.

## Changes Proposed:
- Adds necropoli and (some) dungeon bosses as a part of the scourge invasion event.

## Issues Addressed:

- Makes progress towards https://github.com/azerothcore/azerothcore-wotlk/issues/5786

## SOURCE:

Wrath Classic PTR sniffs.

## Tests Performed:

- Untested.

## How to Test the Changes:

1. Run the query.
2. `.event start 17`
3. Run to various capital cities and locations around the world to observe that necropoli/bosses are spawned. Feel free to use guides (such as [Wowhead's](https://www.wowhead.com/wotlk/guide/scourge-invasion-event-pre-patch-wrath-of-the-lich-king)) to find where they are spawned.

## Known Issues and TODO List:

- [ ] Bosses have no mechanics added in this PR. According to this related [issue](https://github.com/azerothcore/azerothcore-wotlk/issues/5786) and [PR](https://github.com/azerothcore/azerothcore-wotlk/pull/5787), Lady Falther'ess needs some work. [Scorn](https://github.com/azerothcore/azerothcore-wotlk/issues/5750) also likely needs reworking to spawn after Bloodmage Thalnos is defeated.
- [ ] Certain bosses (namely [scholo boss](https://www.wowhead.com/wotlk/npc=14695) and [kara boss](https://www.wowhead.com/wotlk/npc=28194) were not able to be sniffed during the PTR prepatch and thus are not included in this PR unfortunately.
- [ ] Pretty much everything to do with necrotic shards hasn't been touched yet. This includes quests, gobjects, creatures, spells, anything.
- [ ] Flameshockers invading Undercity and Stormwind City are not added in this PR.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
